pkg/controller/resource_sync_task.go

```diff
@@ -176,6 +177,16 @@ func (st *resourceSyncTask) checkAllDependenciesAreReady(res *smith_v1.Resource)
 	return nil
 }
 
+// This 'cache' is bounded in size by the number of plugins we load; we never
+// need to evict. Yes, in some ways it would be nicer to build these schemas
+// when we originally load the plugins, but then we need to have another
+var schemaValidatorCache map[smith_v1.PluginName]*gojsonschema.Schema
```

Comment:
No package-level mutable vars please. This will not work nicely with unit tests, for example.

Reply:
Instead I'd cache the result of the `Describe()` call in the container struct along with the parsed schema. All done at startup time. Why not?